BrowserKit/Sources/WebEngine/WKWebview/Scripts/AdsTelemetryContentScript.swift

```diff
@@ -0,0 +1,151 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Foundation
+import WebKit
+
+public enum EngineBasicSearchProvider: String {
+    case google
+    case duckduckgo
+    case yahoo
+    case bing
+}
+
+public struct EngineSearchProviderModel {
```

Comment:
Those feels more like business logic/Client specific 🤔 Are those the same between Focus and Firefox? Wondering if we can avoid having this inside the WebEngine

Reply:
> Are those the same between Focus and Firefox?

Yes the `AdsTelemetryHelper` is duplicated across both Firefox and Focus and there are some small differences but these regexes and definitions for the `EngineSearchProviderModel` appear to be identical between both. 👍 

> Wondering if we can avoid having this inside the WebEngine

My first inclination was to keep this private to the WebEngine since it seemed specific to how the engine is parsing the results of the Javascript (which will eventually also be part of the WebEngine instead of the Client, AFAIU). But I'm not entirely sure if this needs to be exposed to the Client or not for the handling for `adsClickedOnPage`. I will dig in a bit more into that and figure out how it needs to work. (Or if you have more context on this already and want to chat just LMK.)

Reply:
@lmarceau I moved these out of the WebEngine and into the Client, via delegate callbacks. So the clients can provide the search provider definitions. Since these seem to be common definitions used by both Firefox and Focus, and since we'll probably want them in place for testing, I did also add these to the Sample Browser (in `DefaultAdsTrackerDefinitions.swift`).